### PR TITLE
Support 32K seq length for Qwen2

### PR DESCRIPTION
**Pull request by Zhiwei35**
# What does this PR do?
support the 32k seq length for qwen2 model

### CODE REVIEW

File: optimum/habana/transformers/models/qwen2/modeling_qwen2.py [hunk 1]
```diff
@@ -350,8 +350,15 @@ def pre_attn_forward(
                     attn_output = self.fused_scaled_dot_product_attention(
                         query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
                     )
+            elif q_len > 32768:
+                # first token case1: Long sequence should go to flash_attn_v1 to avoid OOM issue
```
Comment (libinta) [inline optimum/habana/transformers/models/qwen2/modeling_qwen2.py:354]: what about in training case?

**Comment by yafshar** [comment]
@Zhiwei35 can you provide your testing command for this PR?

**Comment by emascarenhas** [comment]
> @Zhiwei35 can you provide your testing command for this PR?

@Zhiwei35 , I tried different options with the examples/trl/ sft.py, but I could not increase the input_seq_len to 32K. please provide use cases for the PR to move it forward.

**Comment by libinta** [comment]
@Zhiwei35  please response to the comments, otherwise, it will delay the integration of this PR.Thx

**Comment by yafshar** [comment]
@Zhiwei35, can you update the missing info for this PR, so we can finish the review and close this PR?

**Comment by libinta** [comment]
@Zhiwei35  can you please update your patch with latest baseline?

**Comment by libinta** [comment]
@Zhiwei35  I will close the PR as no response from you.

**Pull request closed without merging**